uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Listener/AbstractListener.php

```diff
@@ -0,0 +1,25 @@
+<?php
+  abstract class CRM_HRCore_Listener_AbstractListener {
+
+    protected $objectClass;
+    protected $object;
+
+    public function __construct(&$object) {
+      $this->object = &$object;
```

Comment:
not all civicrm hooks pass what you call $object, I think it is more reasonable if you pass such variables at the onHookLisnter level, for example : 


```php
public function onBuildForm(&$object) {
}
```

and remove the $object property